### ISSUE TITLE
BrightcoveExoPlayerVideoView in layout XML breaks IDE preview

### ISSUE THREAD

**Issue by mlagerberg** [BrightcoveOS/android-player-samples#153]
`isInEditMode()` should be used to skip code when the custom view is shown in the preview in Android Studio.

```
java.lang.NullPointerException
	at com.brightcove.player.analytics.Analytics.getDeviceType(Analytics.java:881)
	at com.brightcove.player.analytics.Analytics.<init>(Analytics.java:195)
	at com.brightcove.player.analytics.Analytics.<init>(Analytics.java:185)
	at com.brightcove.player.display.VideoDisplayComponent.<init>(VideoDisplayComponent.java:194)
	at com.brightcove.player.display.ExoPlayerVideoDisplayComponent.<init>(ExoPlayerVideoDisplayComponent.java:282)
	at com.brightcove.player.view.BrightcoveExoPlayerVideoView.createVideoDisplayComponent(BrightcoveExoPlayerVideoView.java:73)
	at com.brightcove.player.view.BaseVideoView.setEventEmitter(BaseVideoView.java:388)
	at com.brightcove.player.view.BaseVideoView.onFinishInflate(BaseVideoView.java:325)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:876)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:72)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:837)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:866)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:72)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:837)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:866)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:72)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:837)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:866)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:72)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:837)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:394)
	at com.android.layoutlib.bridge.impl.RenderSessionImpl.inflate(RenderSessionImpl.java:325)
	at com.android.layoutlib.bridge.Bridge.createSession(Bridge.java:384)
	at com.android.tools.idea.layoutlib.LayoutLibrary.createSession(LayoutLibrary.java:193)
	at com.android.tools.idea.rendering.RenderTask.createRenderSession(RenderTask.java:547)
	at com.android.tools.idea.rendering.RenderTask.lambda$inflate$3(RenderTask.java:681)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by BCjwhisenant** [comment]
Hello,
We are looking into this issue.
Can you share a code sample that is causing the problem?

**Comment by mlagerberg** [comment]
Hi,
here's the simplified code that I can share:

```
<?xml version="1.0" encoding="utf-8"?>
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    android:id="@+id/vg_content"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:orientation="vertical">

    ...

    
            <com.brightcove.player.view.BrightcoveExoPlayerVideoView
                android:id="@+id/brightcove_video_view"
                android:layout_width="match_parent"
                android:layout_height="match_parent"
                android:layout_gravity="center_horizontal|top"
                tools:src="@tools:sample/backgrounds/scenic" />

    ...

</LinearLayout>
```

Using library `com.brightcove.player:exoplayer:4.6+`.

**Comment by BCjwhisenant** [comment]
Thanks very much.

**Comment by sergiomartinez4** [comment]
This issue was fixed in 6.3.1
